src/Stack2nix.hs

```diff
@@ -91,7 +93,7 @@ stack2nix args@Args{..} = do
 
     checkRuntimeDeps :: IO ()
     checkRuntimeDeps = do
-      checkVer "cabal2nix" "2.2.1"
+      checkVer "cabal2nix" "2.7"
```

Comment:
It's not due to this PR, but this check passes for example with cabal2nix 2.5

Reply:
This is fixed in the latest commit. The previous version parser was interpreting the following as `22.7` (first 2 from `cabal2nix`).

```
$ cabal2nix --version
cabal2nix 2.7
```

Now we have a regular expression and a longest match heuristic and test cases to verify a few cases, including the scenario in #67.